test/common/config/config_provider_impl_test.cc

```diff
@@ -369,6 +369,31 @@ TEST_F(ConfigProviderImplTest, DuplicateConfigProto) {
   subscription.onConfigUpdate(untyped_dummy_configs, "1");
 }
 
+// An empty config provider tests on base class' constructor.
+class InlineDummyConfigProvider : public ImmutableConfigProviderBase {
+public:
+  InlineDummyConfigProvider(Server::Configuration::FactoryContext& factory_context,
+                            DummyConfigProviderManager& config_provider_manager,
+                            ConfigProviderInstanceType instance_type)
+      : ImmutableConfigProviderBase(factory_context, config_provider_manager, instance_type,
+                                    ApiType::Full) {}
+  ConfigConstSharedPtr getConfig() const override { return nullptr; }
+  std::string getConfigVersion() const override { return ""; }
+  const Protobuf::Message* getConfigProto() const override { return nullptr; }
+};
+
+TEST_F(ConfigProviderImplTest, AssertionFailureOnIncorrectInstanceType) {
```

Comment:
The test suite should be suffixed with `DeathTest` per https://github.com/google/googletest/blob/master/googletest/docs/advanced.md#death-test-naming.

Reply:
PTAL.